Install MUSIC on mac

Hi there,
I'm using MUSIC to run a multi-simulation with NEST. I'm currently working on ubuntu 14.04 where I was able to install the library without any particular issues. I'd like to reproduce the same environment also on a mac laptop in order to keep working without my lab's desktop. I've been able to install pretty much everything but MUSIC. I got this error:

ld: unknown option: -z
collect2: error: ld returned 1 exit status
make[2]: *** [libmusic.la] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2

Since the default compiler clang coming with the os does not support openmp I wrapped the openmpi installation with gcc4.8 compiler installed with homebrew and I'm using it to compile MUSIC. It should use the same GNU ld of ubuntu so I don't understand  why I keep getting this error. 
Is there a solution for this problem?

Thanks

--- a/autogen.sh
+++ b/autogen.sh
@@ -12,9 +12,16 @@ if test -z "$AUTOMAKE" ; then
  done
 fi
 
+if [ `uname -s` = Darwin ] ; then
+# libtoolize is glibtoolize on OSX
+  LIBTOOLIZE=glibtoolize
+else
+  LIBTOOLIZE=libtoolize
+fi
+
 ./aclocal.sh &&
-echo libtoolize --copy --automake &&
-libtoolize --copy --automake &&
+echo $LIBTOOLIZE --copy --automake &&
+$LIBTOOLIZE --copy --automake &&
 echo autoheader &&
 autoheader &&
 echo autoconf &&

--- a/extras/Makefile.am
+++ b/extras/Makefile.am
@@ -3,5 +3,4 @@
 EXTRA_DIST = music_mpirun
 
 install-data-hook:
-	@INSTALL_PROGRAM@ $(srcdir)/music@LAUNCHSTYLE@ $(DESTDIR)$(bindir)/MUSIC
-
+	@INSTALL_PROGRAM@ $(srcdir)/music@LAUNCHSTYLE@ $(DESTDIR)$(bindir)/MUSIC2

--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -47,7 +47,7 @@ libmusic_la_SOURCES = \
 libmusic_la_HEADERS = music.hh
 libmusic_la_CXXFLAGS = -I$(top_srcdir) @MPI_CXXFLAGS@
 libmusic_la_LDFLAGS = $(top_builddir)/mpidep/libmpidep.la $(top_builddir)/rudeconfig/librudeconfig.la \
-	-version-info 1:0:0 -export-dynamic -Wl,-z,defs @MPI_LDFLAGS@
+	-version-info 1:0:0 -export-dynamic @MPI_LDFLAGS@
 libmusic_ladir = $(includedir)
 
 libmusic_c_la_SOURCES = \
@@ -58,7 +58,7 @@ libmusic_c_la_HEADERS = music-c.h  music/predict_rank-c.h
 libmusic_c_la_CXXFLAGS = -DBUILDING_MUSIC_LIBRARY @MPI_CXXFLAGS@
 libmusic_c_la_CFLAGS = -DBUILDING_MUSIC_LIBRARY @MPI_CFLAGS@
 libmusic_c_la_LDFLAGS = \
-	-version-info 1:0:0 -export-dynamic -Wl,-z,defs \
+	-version-info 1:0:0 -export-dynamic \
 	$(top_builddir)/src/libmusic.la @MPI_LDFLAGS@
 libmusic_c_ladir = $(includedir)
 libmusic_c_la_DEPENDENCIES = libmusic.la

--- a/src/memory.cc
+++ b/src/memory.cc
@@ -19,7 +19,11 @@
 #include <iostream>
 
 extern "C" {
+#ifdef __APPLE__
+#include <malloc/malloc.h>
+#else
 #include <malloc.h>
+#endif
 }
 
 #include "music/memory.hh"

